includes/bp-members/classes/class-bp-members-endpoints.php

```diff
@@ -0,0 +1,402 @@
+			'meta_key'     => $request['meta_key'],
+			'meta_value'   => $request['meta_value'],
+			'user_id'      => $request['user_id'],
+			'type'         => $request['type'],
+			'populate_extras' => $request['populate_extras'] === false ? false: true
+		);
+
+		if ( !empty($args['include']) ) {
+			$args['count_total'] = false;
+		}
+
+		foreach(['user_id', 'meta_key', 'meta_value', 'exclude', 'include'] as $param) {
+			if (array_key_exists($param, $args) && empty($args[$param])) {
+				unset($args[$param]);
+			}
+		}
+		return $args;
+	}
+	/**
+	 * Retrieve activity.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request
+	 *
+	 * @return WP_REST_Request|WP_Error Plugin object data on success, WP_Error otherwise.
+	 */
+	public function get_item( $request ) {
+		$retval     = array();
+		if ( bp_has_members( ['include'=>[(int) $request['id']], 'type'=>'alphabetical'] ) ) :
+			while ( bp_members() ) : bp_the_member();
+				$member = [];
+				$member['user_id'] = bp_get_member_user_id();
+				$member['user_name'] = bp_get_profile_field_data( ['field'=>'Name', 'user_id'=>bp_get_member_user_id()] );
+				$member['user_nicename'] = bp_get_member_user_nicename();
+				$member['user_avatar'] = bp_get_member_avatar();
+				$member['user_permalink'] = bp_get_member_permalink();
+				$member['diary_title'] = bp_get_profile_field_data( ['field'=>'Diary Title', 'user_id'=>bp_get_member_user_id()] );
+				$member['total_friend_count'] = bp_get_member_total_friend_count();
+				$retval = $member;
+			endwhile;
+		endif;
+
+		return new WP_REST_Response( $retval, 200 );
+
+	}
+
+
+	/**
+	 * Check if a given request has access to get information about a specific activity.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return bool
+	 */
+	public function get_item_permissions_check( $request ) {
+		return $this->get_items_permissions_check( $request );
+	}
+
+	/**
+	 * Check if a given request has access to activity items.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return WP_Error|bool
+	 */
+	public function get_items_permissions_check( $request ) {
+		// TODO: handle private activities etc
+		return true;
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param array $activity Activity.
+	 *
+	 * @return array Links for the given plugin.
+	 */
+	protected function prepare_links( $activity ) {
+		$base = sprintf( '/%s/%s/', $this->namespace, $this->rest_base );
+
+		// Entity meta.
+		$links = array(
+			'self'       => array(
+				'href' => rest_url( $base . $activity->id ),
+			),
+			'collection' => array(
+				'href' => rest_url( $base ),
+			),
+			'author'     => array(
+				'href' => rest_url( '/wp/v2/users/' . $activity->user_id ),
+			)
+		);
+
+		if ( $activity->type === 'activity_comment' ) {
+			$links['up'] = array(
+				'href' => rest_url( $base . $activity->item_id ),
+			);
+		}
```

Comment:
might be overkill, but I typically wrap `esc_url_raw()` even on WP functions to be extra safe, eg, `esc_url_raw( rest_url( $base . $activity->item_id ) ),`